src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -4277,6 +4277,122 @@ class StubGenerator: public StubCodeGenerator {
     return (address) start;
   }
 
+  void chacha20_quarter_round(VectorRegister aVec, VectorRegister bVec,
+                          VectorRegister cVec, VectorRegister dVec, VectorRegister tmp_vr) {
+    // a += b, d ^= a, d <<<= 16
+    __ vadd_vv(aVec, aVec, bVec);
+    __ vxor_vv(dVec, dVec, aVec);
+    __ vrol_vwi(dVec, 16, tmp_vr);
+
+    // rev32(dVec, T8H, dVec);
+
+    // c += d, b ^= c, b <<<= 12
+    __ vadd_vv(cVec, cVec, dVec);
+    __ vxor_vv(bVec, bVec, cVec);
+    __ vrol_vwi(bVec, 12, tmp_vr);
+
+    // a += b, d ^= a, d <<<= 8
+    __ vadd_vv(aVec, aVec, bVec);
+    __ vxor_vv(dVec, dVec, aVec);
+    __ vrol_vwi(dVec, 8, tmp_vr);
+
+    // c += d, b ^= c, b <<<= 7
+    __ vadd_vv(cVec, cVec, dVec);
+    __ vxor_vv(bVec, bVec, cVec);
+    __ vrol_vwi(bVec, 7, tmp_vr);
+  }
+
+  /**
+   * int com.sun.crypto.provider.ChaCha20Cipher.implChaCha20Block(int[] initState, byte[] result)
+   *
+   *  Input arguments:
+   *  c_rarg0   - state, the starting state
+   *  c_rarg1   - key_stream, the array that will hold the result of the ChaCha20 block function
+   */
+  address generate_chacha20Block() {
+    Label L_Rounds;
+
+    __ align(CodeEntryAlignment);
+    StubCodeMark mark(this, "StubRoutines", "chacha20Block");
+    address start = __ pc();
+
+    const int states_len = 16;
+    const int step = 4;
+    const Register state = c_rarg0;
+    const Register key_stream = c_rarg1;
+    const Register loop = t0;
+    const Register tmp_addr = t1;
+    const Register length = t2;
+    const Register avl = x28;
+    const Register stride = x29;
+
+    const VectorRegister work_vrs[16] = {
+      v4,  v5,  v6,  v7,  v16, v17, v18, v19,
+      v20, v21, v22, v23, v24, v25, v26, v27
+    };
+    const VectorRegister tmp_vr = v29;
+    const VectorRegister counter = v30;
+
+
+    RegSet saved_regs;
+    __ push_reg(saved_regs, sp);
```

Comment:
You're only using tmp registers it looks like? Is this needed, as the `saved_regs` set is empty?